java-checks-test-sources/src/main/java/checks/spring/NullableInjectedFieldsHaveDefaultValueCheck.java

```diff
@@ -0,0 +1,46 @@
+package checks.spring;
+
+import javax.annotation.Nullable;
+import org.springframework.beans.factory.annotation.Value;
+
+public class NullableInjectedFieldsHaveDefaultValueCheck {
+  @Nullable
+  @Value("${my.property}") // Noncompliant [[sc=3;ec=27;secondary=-1]] {{Provide a default null value for this field.}}
+  private String myProperty;
+
+  @Value("${my.property}") // Noncompliant [[sc=3;ec=27;secondary=+1]] {{Provide a default null value for this field.}}
+  @Nullable
+  private String myPropertyWithReversedAnnotations;
+
+  @Nullable
+  @Value(value = "${my.property}") // Noncompliant [[sc=3;ec=35]]
+  private String myOtherProperty;
+
+  @Nullable
+  @Value("${my.property}") // Noncompliant [[sc=3;ec=27;quickfixes=literalfix]]
+  private String myPropertyToFix;
+  // fix@literalfix {{Set null as default value}}
+  // edit@literalfix [[sc=10;ec=26]] {{"${my.property:#{null}}"}}
+
+  @Nullable
+  @Value(value = "${my.property}") // Noncompliant [[sc=3;ec=35;quickfixes=argumentfix]]
+  private String myPropertyToFixOnNamedArgument;
+  // fix@argumentfix {{Set null as default value}}
+  // edit@argumentfix [[sc=18;ec=34]] {{"${my.property:#{null}}"}}
+
+  private static final String WEIRD = "${my.property:#{null}}";
+  @Nullable
+  @Value(value = WEIRD) // FN compliant
```

Comment:
Why is this a FN? Isn't this just compliant? Or should the `WEIRD` field not contain a default value? That would be a good test to have and we should be able to detect it.

Reply:
My mistake, I modified the example at some point and made it compliant when it wasn't. I will double the example with a non-compliant one now